pkg/gitprovider/gitnessclient/gitness_client.go

```diff
@@ -353,6 +347,89 @@ func (g *GitnessClient) GetDefaultBranch(url string) (*string, error) {
 	return &repo.DefaultBranch, nil
 }
 
+func (g *GitnessClient) CreateWebhook(repoId string, namespaceId string, callbackURL string) (*Webhook, error) {
+	webhookEndpoint, parseErr := g.BaseURL.Parse(fmt.Sprintf("/api/v1/repos/%s/webhooks", url.PathEscape(namespaceId+"/"+repoId)))
+	if parseErr != nil {
+		return nil, parseErr
+	}
+	webhookData := map[string]interface{}{
```

Comment:
These should be parameters of the function.

Create a new struct `CreateWebhookParams` that will take all the params that can then be passed from the calling function and marshaled.

Reply:
@the-johnwick you didn't implement the required changes for this. The `CreateWebhook` method of the client needs to take all those params as arguments.

E.g. https://github.com/daytonaio/daytona/blob/main/pkg/gitprovider/github.go#L418